softwareComponents/shapeReconfig/src/hashing.cpp

```diff
@@ -0,0 +1,98 @@
+#include <shapeReconfig/hashing.hpp>
+
+namespace rofi::shapereconfig {
+
+using namespace rofi::configuration;
+
+size_t combineHash( size_t h1, size_t h2 ) 
+{ 
+    return h1 ^ (h2 << 1);
+}
+
+// If converted to degrees and rounded, equal degree has same hash
+size_t hashRad( float rads )
+{
+    return std::hash<int>{}( static_cast<int>( round( rads * 100 ) ) );
+}
+
+size_t hashJoint( const Joint& j )
+{
+    size_t h1 = std::accumulate( j.jointLimits().begin(), j.jointLimits().end(), 0, 
+        []( size_t acc, const std::pair< float, float >& lims ){ 
+            size_t h = combineHash( hashRad( lims.first ), hashRad( lims.second ) );
+            return combineHash( acc, h );
+        } );
+
+    size_t h2 = std::accumulate( j.positions().begin(), j.positions().end(), 0, 
+        []( size_t acc, const float& pos ){ 
+            return combineHash( acc, hashRad( pos ) ); 
+        } );
+
+    return combineHash( h1, h2 );
+}
+
+size_t hashRoficomJoint( const RoficomJoint& rj )
+{
+    size_t res = hashJoint( rj );
+    res = combineHash( res, std::hash<size_t>{}( static_cast<size_t>( rj.orientation ) ) );
+    res = combineHash( res, std::hash<size_t>{}( static_cast<size_t>( rj.sourceModule ) ) );
+    res = combineHash( res, std::hash<size_t>{}( static_cast<size_t>( rj.destModule ) ) );
+    res = combineHash( res, std::hash<int>{}( rj.sourceConnector ) );
+    res = combineHash( res, std::hash<int>{}( rj.destConnector ) );
+    return res;
+}
+
+size_t hashComponentJoint( const ComponentJoint& cj )
+{
+    size_t res = hashJoint(*cj.joint);
+    res = combineHash( res, std::hash<int>{}( cj.sourceComponent ));
+    res = combineHash( res, std::hash<int>{}( cj.destinationComponent ));
+    return res;
+}
+
+size_t hashModule( const Module& mod )
+{
+    size_t res = std::hash<size_t>{}(static_cast<size_t>(mod.type));
+    res = combineHash( res, std::hash<int>{}(mod.getId()) );
+    size_t h = std::accumulate( mod.joints().begin(), mod.joints().end(), 0, 
+        []( size_t acc, const ComponentJoint& cj ){ 
+            return combineHash( acc, hashComponentJoint( cj ) ); 
+        } );
+    return combineHash( res, h );
+}
+
+size_t RofiWorldHash::operator()( const RofiWorld& rw ) const
+{
+    size_t h1 = std::accumulate( rw.modules().begin(), rw.modules().end(), 0, 
```

Comment:
This will convert everything to int since init is of type int (use `size_t{}`, `size_t( 0 )`, `size_t{ 0 }` or similar.